server/chat-plugins/smogtours.ts

```diff
@@ -0,0 +1,447 @@
+			};
+			if (isEdit) {
+				const index = section.tours.findIndex(t => t.id === tour.id);
+				if (index < 0) {
+					return this.popupReply(`Tour not found. Create one first.`);
+				}
+				section.tours.splice(index, 1);
+			}
+			section.tours.push(tour);
+			saveTours();
+			this.refreshPage(`tournaments-add`);
+		},
+		end(target, room, user, connection) {
+			const [sectionID, tourID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !tourID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) return this.popupReply(`Invalid section: "${sectionID}"`);
+			const idx = section.tours.findIndex(t => t.id === tourID);
+			const title = section.tours[idx].title;
+			if (idx < 0) {
+				return this.popupReply(`Tour with ID "${tourID}" not found.`);
+			}
+			section.tours.splice(idx, 1);
+			for (const page of connection.openPages || []) {
+				if (page.startsWith('tournaments-')) this.refreshPage(page);
+			}
+			this.popupReply(`Tour "${title}" ended.`);
+		},
+		whitelist(target, room, user) {
+			this.checkCan('rangeban');
+			const [sectionID, targetID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !targetID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) {
+				return this.errorReply(`Invalid section ID: "${sectionID}". Valid IDs: ${Object.keys(tours).join(', ')}`);
+			}
+			if (section.whitelist?.includes(targetID)) {
+				return this.errorReply(`That user is already whitelisted on that section.`);
+			}
+			if (!section.whitelist) section.whitelist = [];
+			section.whitelist.push(targetID);
+			this.privateGlobalModAction(
+				`${user.name} whitelisted ${targetID} to manage tours for the ${section.title} section`
+			);
+			this.globalModlog('TOUR WHITELIST', targetID);
+			saveTours();
+		},
+		unwhitelist(target, room, user) {
+			this.checkCan('rangeban');
+			const [sectionID, targetID] = target.split(',').map(toID).filter(Boolean);
+			if (!sectionID || !targetID) {
+				return this.parse(`/help smogtours`);
+			}
+			const section = tours[sectionID];
+			if (!section) {
+				return this.errorReply(`Invalid section ID: "${sectionID}". Valid IDs: ${Object.keys(tours).join(', ')}`);
+			}
+			const idx = section.whitelist?.indexOf(targetID);
+			if (!section.whitelist || !idx || idx < 0) {
+				return this.errorReply(`${targetID} is not whitelisted in that section.`);
+			}
+			section.whitelist?.splice(idx, 1);
+			if (!section.whitelist.length) {
+				delete section.whitelist;
+			}
+			this.privateGlobalModAction(
+				`${user.name} removed ${targetID} from the tour management whitelist for the ${section.title} section`
+			);
+			this.globalModlog('TOUR UNWHITELIST', targetID);
+			saveTours();
+		},
+		view() {
+			return this.parse(`/join view-tournaments-all`);
+		},
+	},
+	smogtourshelp: [
```

Comment:
"smogtours" makes me think of smogtours.psim.us; maybe "forum tours" would be a better name?

Reply:
I would prefer not - this is not meant for _exclusively_ forums.